src/components/ScheduleGenerate/ScheduleGenerateSideBar.vue

```diff
@@ -90,13 +91,15 @@ export default defineComponent({
     requirements: ReqCourses[];
     isConfirmationOpen: boolean;
     confirmationText: string;
-    creditLimit: number;
+    creditLimit?: number;
```

Comment:
is there a reason we make creditLimit an optional prop?

Reply:
it's undefined initially when the user hasn't put anything in yet